File: ggukgguk-frontend/src/api/member.js
import apiFactory from "./apiFactory"
import store from '../store'

const axios = apiFactory.getInstance();

export default {
    // 회원 정보 조회
    memberInformation(memberId) {
        console.log(memberId);
        return axios.get(`/member/${memberId}`,
            {
                headers: {
                    Authorization: `Bearer ${store.getters['auth/accessToken']}`
                }
            });
    },

    // 친구 목록 조회
    getFriendList() {
        return axios.get(`/friend/list`);
    },

    // 회원 정보 수정
    memberModify({ memberId, memberPw, memberName, memberNickname,
        memberEmail, memberPhone, memberBirth, memberAllowEmail }) {
        return axios.put(`/member/${memberEmail}`,
            {
                memberId, memberPw, memberName, memberNickname,
                memberEmail, memberPhone, memberBirth, memberAllowEmail
            },
            {
                headers: {
                    Authorization: `Bearer ${store.getters['auth/accessToken']}`
                }
            });
    }
} 

File: ggukgguk-backend/ggukgguk/src/main/java/com/ggukgguk/api/member/vo/Member.java
package com.ggukgguk.api.member.vo;

import java.util.Date;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.ggukgguk.api.auth.enums.MemberRoleEnum;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@AllArgsConstructor
@Data
public class Member {
	private String memberId;
	private String memberPw;
	private String memberName;
	@JsonFormat(pattern = "yyyy-MM-dd")
	private Date memberBirth;
	private String memberNickname;
	private String memberEmail;
	private String memberPhone;
	@JsonFormat(pattern = "yyyy-MM-dd'T'hh:mm:ss.SSSZ", timezone = "Asia/Seoul")
	private Date memberCreatedAt;
	private boolean memberActivated;
	private String memberAuthority;
	private String memberSocial;
	private boolean memberAllowEmail;
	
	public String getRole() {
		return MemberRoleEnum.valueOfLabel(memberAuthority).label();
	}
}
